### PR TITLE
Adding backstage configuration - catalog-info.yaml

### PR DESCRIPTION
**Pull request by lokeshrangineni**
Adding backstage configuration - catalog-info.yaml

### Testing done

```[tasklist]
### Submitter checklist
- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or Jira
- [x] Link to relevant pull requests, esp. upstream and downstream changes
- [x] Ensure you have provided tests - that demonstrates feature works or fixes the issue
```

### CODE REVIEW

File: catalog-info.yaml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+apiVersion: backstage.io/v1alpha1
+kind: Component
+metadata:
+  annotations:
+    backstage.io/kubernetes-id: redhat-dependency-analytics-plugin
+    github.com/project-slug: RHEcosystemAppEng/redhat-dependency-analytics-plugin
+    github.com/project-readme-path: README.md
+    backstage.io/view-url: https://github.com/RHEcosystemAppEng/redhat-dependency-analytics-plugin/blob/main/catalog-info.yaml
```
Comment (ruromero) [inline catalog-info.yaml]:
```suggestion
    backstage.io/view-url: https://github.com/jenkinsci/redhat-dependency-analytics-plugin/blob/main/catalog-info.yaml
```

File: catalog-info.yaml [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    backstage.io/edit-url: https://github.com/RHEcosystemAppEng/redhat-dependency-analytics-plugin/blob/main/catalog-info.yaml
```
Comment (ruromero) [inline catalog-info.yaml]:
```suggestion
    backstage.io/edit-url: https://github.com/jenkinsci/redhat-dependency-analytics-plugin/blob/main/catalog-info.yaml
```

File: catalog-info.yaml [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    backstage.io/source-location: url:https://github.com/RHEcosystemAppEng/redhat-dependency-analytics-plugin
```
Comment (ruromero) [inline catalog-info.yaml]:
```suggestion
    backstage.io/source-location: url:https://github.com/jenkinsci/redhat-dependency-analytics-plugin
```

File: catalog-info.yaml [hunk 4]
```diff
@@ -0,0 +1,17 @@
+apiVersion: backstage.io/v1alpha1
+kind: Component
+metadata:
+  annotations:
+    backstage.io/kubernetes-id: redhat-dependency-analytics-plugin
+    github.com/project-slug: RHEcosystemAppEng/redhat-dependency-analytics-plugin
```
Comment (ruromero) [inline catalog-info.yaml]:
```suggestion
    github.com/project-slug: jenkinsci/redhat-dependency-analytics-plugin
```

**Review by ruromero — CHANGES_REQUESTED** [review]

**Review by ruromero — APPROVED** [review]

**Comment by bmozaffa** [comment]
@olavtar on how to merge

**Pull request merged**